lib/src/payments/p2ms.dart

```diff
@@ -0,0 +1,217 @@
+import '../utils/constants/op.dart';
+import 'package:meta/meta.dart';
+import '../utils/script.dart' as bscript;
+import '../models/networks.dart';
+import 'dart:typed_data';
+import 'package:bip32/src/utils/ecurve.dart' show isPoint;
+
+class P2MS {
+  P2MSData data;
+  NetworkType network;
+  List<dynamic> _chunks;
+  Map _temp;
+  bool _isDecoded;
+
+  P2MS({@required data}) {
+    this.data = data;
+    this.network = network ?? bitcoin;
+    this._isDecoded = false;
+    this._temp = {};
+    _init();
+  }
+  void _init() {
+    _enoughInformation(data);
+    _extraValidation(data.options);
+    _setNetwork(this.network);
+    _extendedValidation();
+  }
+
+  void _enoughInformation(data) {
+    if (data.input == null &&
+        data.output == null &&
+        !((data.pubkeys  != null) && (data.m != null)) &&
+        data.signatures == null) {
+      throw new ArgumentError('Not enough data');
+    }
+  }
+  void _setNetwork(network){
+      _temp['network'] == network;
+  }
+  void _extraValidation(options) {
+    if(data.options == null){data.options = {};}
+    data.options['validate'] = true;
+  }
+
+  void _extendedValidation(){
+    if(data.options['validate']==true){
+      _check();
+    }
+    _assignVariables();
+  }
+   void _decode(output){
+     
+    if(_isDecoded) {return;}
+    else{
+      _isDecoded = true;
+      _chunks = bscript.decompile(output);
+      _temp['m'] = _chunks[0] - OPS['OP_RESERVED'];
```

Comment:
Please use P2MSData, dynamic some cases cause dangerous

Reply:
Should I add a type to chunks?